src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobLister.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.reform.blobrouter.services.storage;
+
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.models.BlobItem;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.blobrouter.config.ServiceConfiguration;
+import uk.gov.hmcts.reform.blobrouter.model.out.BlobInfo;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.blobrouter.util.TimeUtils.toLocalTimeZone;
+
+@Component
+public class BlobLister {
+
+    private final BlobServiceClient storageClient;
+    private final ServiceConfiguration serviceConfiguration;
+
+    public BlobLister(BlobServiceClient storageClient, ServiceConfiguration serviceConfiguration) {
```

Comment:
normally it needs `enable configuration properties`. even though it is enabled somewhere else but instantiation of beans can happen in any order and lazy driven too